<launch>
    <arg name="car_id" value="0"/>

    <arg name="init_x" value="0"/>
    <arg name="init_y" value="0"/>
    <arg name="init_yaw" value="0"/> 
    <arg name="init_z" value="0.5"/>
    <arg name="lidar_z" value="0.25"/> 

    <arg name="odom_topic" value="/odom"/>

    <arg name="cmd_topic" value="/cmd"/>

    <!-- laser simulator -->
    <node pkg="laser_simulator" type="laser_sim_node" name="car_$(arg car_id)_laser_simulator" output="screen">
        <param name="car_id" value="$(arg car_id)" />
        <!-- <param name="map_topic" value="/map_generator/global_cloud"/> -->
        <param name="map_topic" value="/global_cloud"/>
        <param name="odom_topic" value="$(arg odom_topic)"/>
        <param name="laser_pcd_topic" value="car_$(arg car_id)_local_pointcloud"/>
        <param name="laser_range_topic" value="car_$(arg car_id)_laser_range" />
        <param name="sensing_horizon" value="10.0"/>
        <param name="sensing_rate" value="10"/>
        <param name="pc_resolution" value="0.05"/>
        <param name="vtc_laser_line_num" value="32"/>
        <param name="vtc_laser_range_dgr" value="30.0"/>
        <param name="hrz_limited" value="false"/>
        <param name="hrz_laser_range_dgr" value="360.0"/>
        <param name="hrz_laser_line_num" value="360"/>
        <param name="use_resolution_filter" value="true"/>
    </node>

    <!-- kimatic simulator -->
    <node pkg="kimatic_simulator" name="car_$(arg car_id)_ugv_kimatic_simulator" type="kimatic_simulator_node" output="screen">
        <param name="map_name" value="hill.pcd" />
        <param name="init_x" value="$(arg init_x)"/>
        <param name="init_y" value="$(arg init_y)"/>
        <param name="init_z" value="$(arg init_z)"/>
        <param name="init_yaw" value="$(arg init_yaw)"/>
        <param name="lidar_z" value="$(arg lidar_z)"/>
        <remap from="~command" to="$(arg cmd_topic)"/>
        <remap from="~odometry" to="$(arg odom_topic)"/>
        <remap from="~marker" to="$(arg odom_topic)_marker"/>
    </node>

    <node name="keyboard_ctrl" pkg="kimatic_simulator" type="keyboard_ctrl.py" output="screen" />

    <node name="rviz" pkg="rviz" type="rviz" args="-d $(find kimatic_simulator)/rviz/default.rviz" required="true" />
    
</launch>
